app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -48,34 +53,103 @@ import org.mozilla.fenix.wallpapers.Wallpaper
  * The screen for controlling settings around Wallpapers. When a new wallpaper is selected,
  * a snackbar will be displayed.
  *
- * @param wallpapers Wallpapers to add to grid.
+ * @param wallpaperGroups Wallpapers groups to add to grid.
  * @param selectedWallpaper The currently selected wallpaper.
  * @param defaultWallpaper The default wallpaper.
  * @param loadWallpaperResource Callback to handle loading a wallpaper bitmap. Only optional in the default case.
  * @param onSelectWallpaper Callback for when a new wallpaper is selected.
+ * @param onLearnMoreClick Callback for when the learn more action is clicked from the group description.
  */
 @SuppressLint("UnusedMaterialScaffoldPaddingParameter")
 @Composable
 @Suppress("LongParameterList")
 fun WallpaperSettings(
-    wallpapers: List<Wallpaper>,
+    wallpaperGroups: Map<Wallpaper.Collection, List<Wallpaper>>,
     defaultWallpaper: Wallpaper,
     loadWallpaperResource: suspend (Wallpaper) -> Bitmap?,
     selectedWallpaper: Wallpaper,
     onSelectWallpaper: (Wallpaper) -> Unit,
+    onLearnMoreClick: (String) -> Unit,
 ) {
     Column(
         modifier = Modifier
+            .fillMaxWidth()
+            .fillMaxHeight()
             .verticalScroll(rememberScrollState())
             .background(color = FirefoxTheme.colors.layer1),
     ) {
-        WallpaperThumbnails(
-            wallpapers = wallpapers,
-            defaultWallpaper = defaultWallpaper,
-            loadWallpaperResource = loadWallpaperResource,
-            selectedWallpaper = selectedWallpaper,
-            onSelectWallpaper = { updatedWallpaper -> onSelectWallpaper(updatedWallpaper) },
-        )
+        wallpaperGroups.forEach { (collection, wallpapers) ->
+            if (wallpapers.isNotEmpty()) {
+                WallpaperGroupHeading(
+                    collection,
+                    onLearnMoreClick,
+                )
+                WallpaperThumbnails(
```

Comment:
Add a new line before to visually separate the 2 Composable for easier readability.